File: 백준/Silver/10211. Maximum Subarray/Maximum Subarray.py
import sys
input = sys.stdin.readline

t = int(input().strip())
answer = []
for tc in range(0,t): 
  n = int(input().strip())
  numbers = list(map(int,input().strip().split(" ")))
  dp = numbers[:]

  for i in range(1,n):
    if (dp[i] + dp[i - 1] > dp[i]):
      dp[i] += dp[i - 1]
  answer.append(str(max(dp)))

print('\n'.join(answer))

File: 백준/Silver/10211. Maximum Subarray/Maximum Subarray.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")

const t = Number(input[0])
for (let tc = 0; tc < t; tc += 1) {
  const n = Number(input[tc * 2 + 1])
  const numbers = input[tc * 2 + 2].split(" ").map(Number)
  const dp = [...numbers]

  for (let i = 1; i < n; i += 1) {
    if (dp[i] + dp[i - 1] > dp[i]) {
      dp[i] += dp[i - 1]
    }
  }
  console.log(Math.max(...dp))
}
